static/js/compose.js

```diff
@@ -1332,6 +1333,7 @@ export function initialize() {
 
     $("#compose-textarea").on("focus", () => {
         compose_actions.update_placeholder_text();
+        drafts.restore_last_draft_based_on_compose_state();
```

Comment:
We are restoring draft here instead of when `#stream_message_recipient_stream`, `#stream_message_recipient_topic` and `#private_message_recipient` are blurred because it creates an issue when topic A and topic B has a similar name(e.g: `topic`, `topic 1`) and we have a draft for topic A and when we type topic A in the input box and click on topic B from typeahead it loads topic A's draft because input box loses focus.

Below gif better explains the issue,
![issue](https://user-images.githubusercontent.com/63820270/120113826-e620d300-c199-11eb-8a31-c13a3bf05996.gif)